File: app/src/main/java/com/example/lk/kotlinmvp/uitls/DataRecycleUtils.java
package com.example.lk.kotlinmvp.uitls;

import android.databinding.BindingAdapter;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;

import com.example.lk.kotlinframework.adapter.base.BaseAdapterIml;

import com.example.lk.kotlinmvp.R;
import com.example.lk.kotlinmvp.mvp.m.bean.FindBean;

import java.util.List;

/**
 * Created by lk on 2018/6/11.
 */

public class DataRecycleUtils {
    @BindingAdapter("find")
    public static void setfindadapter(RecyclerView recyclerView, List<FindBean> data) {
        if (data == null) {
            return;
        }
        recyclerView.setLayoutManager(new GridLayoutManager(recyclerView.getContext(), 2));
//        recyclerView.setLayoutManager(new LinearLayoutManager(recyclerView.getContext(), LinearLayoutManager.VERTICAL, false));
        int layout = R.layout.find_item;
//        for (int i = 0; i < data.size(); i++) {
//            Log.e("test", "name--" + data.get(i).getName() + "--bgPicture--" + data.get(i).getBgPicture());
//        }

        BaseAdapterIml adapter = new BaseAdapterIml(recyclerView.getContext(), data, layout);
        recyclerView.setAdapter(adapter);
    }

}

File: app/src/main/java/com/example/lk/kotlinmvp/uitls/DataRecycleUtils.kt
package com.example.lk.kotlinmvp.uitls

import android.databinding.BindingAdapter
import android.support.v7.widget.GridLayoutManager
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log

import com.example.lk.kotlinframework.adapter.base.BaseAdapterIml

import com.example.lk.kotlinmvp.R
import com.example.lk.kotlinmvp.mvp.m.bean.FindBean

/**
 * Created by lk on 2018/6/11.
 */

class DataRecycleUtils {
    companion object {
        @BindingAdapter("find")
        @JvmStatic
        fun setfindadapter(recyclerView: RecyclerView, data: MutableList<FindBean>?) {
            if (data == null) {
                return
            }
            recyclerView.layoutManager = GridLayoutManager(recyclerView.context, 2)
            val layout = R.layout.find_item
            val adapter = BaseAdapterIml(recyclerView.context, data, layout)
            recyclerView.adapter = adapter
        }
    }
}
